sg/deps.go

```diff
@@ -22,6 +23,20 @@ func Deps(ctx context.Context, functions ...interface{}) {
 	var wg sync.WaitGroup
 	for i, f := range checkedFunctions {
 		i, f := i, f
+
+		dependencies := getDependencies(ctx)
+		for _, dependency := range dependencies {
+			if dependency.ID() == f.ID() {
+				depNames := make([]string, 0, len(dependencies))
+				for _, dependency := range dependencies {
+					depNames = append(depNames, dependency.Name())
+				}
```

Comment:
```suggestion
				depNames := make([]string, len(dependencies))
				for i, dependency := range dependencies {
					depNames[i] = dependency.Name()
				}
```

Dunno if you can do that but is usually a good idea when you want to "copy" a slice :)

Reply:
Ah, sure, I'll change! Not at all happy with the code here, loop in a loop etc, but the inner loop is only executed when panicking later so maybe it's fine 😅